### ISSUE TITLE
Swiften won't connect to the XMPP server with TLS enabled, but without StartTLS

### ISSUE THREAD

**Issue by jw0k** [swift/swift#116]
We have an XMPP server hidden behind the balancer. The balancer itself provides the TLS encryption. However it does not support upgrading a plaintext connection to an encrypted connection (a.k.a. StartTLS or opportunistic TLS). I suppose for this reason Swiften cannot connect to our server. Is it possible to set up Swiften in such a way that the connection is encrypted for the very beginning (without StartTLS command)? It is possible, e.g. in Psi (the option is called "Legacy SSL").

**Comment by Kev** [comment]
Hi. Direct TLS isn't something that Swiften can currently use. It is a thing it would be useful, but I don't think any of us will have time to work on it soon, I'm afraid.

**Comment by jw0k** [comment]
@Kev That's OK. In the meantime I managed to implement it myself. Are you interested in the pull request?

**Comment by Kev** [comment]
Yes please. Contributions are under the BSD license, with a note to that effect in the commit message (e.g. https://github.com/swift/swift/commit/110eb87e848b85dd74a6f19413c775520a75ea35 ) - if you're happy with that, please open a PR and we can go from there.

**Comment by jw0k** [comment]
OK then, I've opened a PR: https://github.com/swift/swift/pull/117

**Comment by linuxpham** [comment]
XMPP test server : xmpp.jp


*** Aborted at 1602556412 (Unix time, try 'date -d @1602556412') ***
*** Signal 11 (SIGSEGV) (0x1b803890e74) received by PID 9600 (pthread TID 0x7f87f5111700) (linux TID 9726) (code: address not mapped to object), stack trace: ***

==8403== by 0x506389F: ??? (in /lib/x86_64-linux-gnu/libpthread-2.27.so)
==8403== by 0x1B803890E73: ???
==8403== by 0x2338DAE: SSL_CTX_new (in /var/www/heratap/HeraChat/HAChatServer)
==8403== by 0x1F74AF4: Swift::(anonymous namespace)::createSSL_CTX(Swift::TLSContext::Mode) (OpenSSLContext.cpp:87)
==8403== by 0x1F77B64: Swift::OpenSSLContext::OpenSSLContext(Swift::TLSOptions const&, Swift::TLSContext::Mode) (OpenSSLContext.cpp:109)
==8403== by 0x1EA3B0F: make_unique<Swift::OpenSSLContext, const Swift::TLSOptions&, Swift::TLSContext::Mode&> (unique_ptr.h:857)
==8403== by 0x1EA3B0F: Swift::OpenSSLContextFactory::createTLSContext(Swift::TLSOptions const&, Swift::TLSContext::Mode) (OpenSSLContextFactory.cpp:25)
==8403== Address 0x1b803890000 is not stack'd, malloc'd or (recently) free'd
